src/Aspire.Hosting.Azure.ServiceBus/ServiceBusHealthCheck.cs

```diff
@@ -0,0 +1,60 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Azure.Messaging.ServiceBus;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace Aspire.Hosting.Azure.ServiceBus;
+
+/// <summary>
+/// The standard ServiceBus health check is not sufficient since it requires predefined knowledge that there is at least one queue or one topic.
+/// At the time we register the health check, we may not know if there are any of each.
+/// </summary>
+internal sealed class ServiceBusHealthCheck : IHealthCheck
+{
+    private readonly Func<ServiceBusClient> _clientFactory;
+    private readonly Func<IEnumerable<string>> _queueNamesFactory;
+    private readonly Func<IEnumerable<string>> _topicNamesFactory;
+
+    private ServiceBusClient? _serviceBusClient;
+
+    public ServiceBusHealthCheck(Func<ServiceBusClient> clientFactory, Func<IEnumerable<string>> queueNamesFactory, Func<IEnumerable<string>> topicNamesFactory)
+    {
+        ArgumentNullException.ThrowIfNull(clientFactory);
+        ArgumentNullException.ThrowIfNull(queueNamesFactory);
+        ArgumentNullException.ThrowIfNull(topicNamesFactory);
+
+        _clientFactory = clientFactory;
+        _queueNamesFactory = queueNamesFactory;
+        _topicNamesFactory = topicNamesFactory;
+    }
+
+    /// <inheritdoc />
+    public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+    {
+        // ServiceBusAdministrationClient is not available in the emulator
+
+        try
+        {
+            _serviceBusClient ??= _clientFactory();
```

Comment:
It doesn't really pay to cache this client in a field because the health check instance is newly created every time the health check is executed. The IHealthCheck instances aren't reused.

Reply:
Didn't know, thanks. I assume that's why the one in the package is using a helper with a static instance. Should we cache it in the factory? Is it fine to keep the cache in the factory itself then? Otherwise I can try to get it from the service provider.

Reply:
The "normal" route we take is to put the client in DI, and then it gets cached there. But in the Hosting libraries that doesn't seem to be the pattern. If this is expensive to create, we can cache it in a closure in the AzureServiceBusExtensions. I'm doing that in the RabbitMQ updates I'm working on.

Or if it isn't that expesive, we can just create a new client every check.